### PR TITLE
Week 6: Scientific Paper Proposal

### PR DESCRIPTION
**Pull request by dvavd**
# Assignment Proposal

## Title

Implementing and Automating Security Scanning to
a DevSecOps CI/CD Pipeline

## Names and KTH ID

  - Prerna Gupta (prerna@kth.se)
  - David Streuli (streuli@kth.se)

## Deadline

- Week 6

## Category

- Scientific paper

## Description

The paper explores the automation of security scanning focusing on containerised applications. We'll explain how integration SAST and DAST tools improves the security of containerized applications by finding vulnerabilities soon and automating their removal from the CI/CD process, the effectiveness of deploying and automating security scanning in DevSecOps pipeline with Snyk and StackHawk tools and methodologies used for detecting vulnerabilities. We conclude the presentation with future scope. The paper can be found here: https://ieeexplore.ieee.org/abstract/document/10235015

**Relevance**

In this paper, a DevSecOps CI/CD pipeline integrated dynamic security testing strategy to address the special requirements of securing containerised applications. The model features early vulnerability detection and push left practices with automated remediation using tools such as Snyk for Static Application Security Testing (SAST) and StackHawk for Dynamic Application Security Testing (DAST), all during the software development lifecycle (SDLC).

### CODE REVIEW

File: contributions/scientific-paper/week6/streuli-prerna/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+- Week 6
+
+## Category
+
+- Scientific paper
+
+## Description
+
+The paper explores the automation of security scanning focusing on containerised applications. We'll explain how integration SAST and DAST tools improves the security of containerized applications by finding vulnerabilities soon and automating their removal from the CI/CD process, the effectiveness of deploying and automating security scanning in DevSecOps pipeline with Snyk and StackHawk tools and methodologies used for detecting vulnerabilities. We conclude the presentation with future scope. The paper can be found here: https://ieeexplore.ieee.org/abstract/document/10235015
```
Comment (algomaster99) [inline contributions/scientific-paper/week6/streuli-prerna/README.md]:
```suggestion
The paper explores the automation of security scanning focusing on containerised applications. We'll explain how integrating SAST and DAST tools improves the security of containerized applications by finding vulnerabilities soon and automating their removal from the CI/CD process, the effectiveness of deploying and automating security scanning in DevSecOps pipeline with Snyk and StackHawk tools and methodologies used for detecting vulnerabilities. We conclude the presentation with future scope. The paper can be found here: https://ieeexplore.ieee.org/abstract/document/10235015
```

**Review by algomaster99 — CHANGES_REQUESTED** [review]
Hi! Good proposal. Could you also include a bit of background about SAST and DAST? Once you will do the changes, I will merge.

**Comment by dvavd** [comment]
@algomaster99 it's updated

**Comment by algomaster99** [comment]
Failed for similar reason :(
https://github.com/KTH/devops-course/pull/2519, @Deee92

**Comment by javierron** [comment]
@algomaster99 @Deee92 CI is fixed

**Comment by algomaster99** [comment]
@javierron what was the reason?

**Pull request merged**